src/core/meshcop/border_agent.hpp

```diff
@@ -87,15 +88,30 @@ class BorderAgent : public InstanceLocator, private NonCopyable
      * be published in the MeshCoP mDNS service as the `id` TXT value for the client to identify this
      * Border Router/Agent device.
      *
-     * @param[out]   aId      A pointer to buffer to receive the ID.
-     * @param[inout] aLength  Specifies the length of `aId` when used as input and receives the length
-     *                        actual ID data copied to `aId` when used as output.
+     * @param[out] aId  receives the Border Agent ID.
+     *
+     * @retval OT_ERROR_NONE  If successfully retrieved the Border Agent ID.
+     * @retval ...            If failed to retrieve the Border Agent ID.
+     *
+     */
+    Error GetId(Id &aId);
+
+    /**
+     * Sets the Border Agent ID.
+     *
+     * The Border Agent ID will be saved in persistent storage and survive reboots. It's required to
+     * set the ID only once after factory reset.
```

Comment:
> It's required to set the ID only once after factory reset.

We do allow it to be set again after setting it, right?

Reply:
Yes, we allow it. But it is generally a programming error for a device to use two different IDs without factory reset

Reply:
Thanks. SGTM